homeassistant/components/ring/entity.py

```diff
@@ -47,10 +47,16 @@ class DeprecatedInfo:
 
 
 @dataclass(frozen=True, kw_only=True)
-class RingEntityDescription(EntityDescription):
+class RingEntityDescription(EntityDescription, Generic[RingDeviceT]):
     """Base class for a ring entity description."""
 
     deprecated_info: DeprecatedInfo | None = None
+    exists_fn: Callable[[RingDeviceT], bool]
+    dynamic_exists_fn: Callable[[RingDeviceT], bool] | None = None
```

Comment:
What's the difference between dynamic and normal?

Reply:
The presence of a dynamic_exists function will cause the entities to be automatically cleaned up when the integration is loaded. The regular exists function doesn't trigger any clean up. The idea behind splitting them was to avoid checking the entity registry for every single entity regardless of whether it's expected to disappear or not.

Reply:
But why do we have to check the entity registry every time?